drivers/hmis_external_apis/app/controllers/hmis_external_apis/referrals_controller.rb

```diff
@@ -0,0 +1,29 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis
+  class ReferralsController < BaseController
+    before_action :authorize_request
+    skip_before_action :authenticate_user!
+    prepend_before_action :skip_timeout
+
+    def create
```

Comment:
So we'd use an exception to handle validation failures? That might be necessary but I suggest we try a normal control flow first. Then exceptions would be reserved for errors and we could just leave them unhandled (ensure instead of rescue). Something like this:

```ruby
    def create
      # with_logging do
      #  ...
      # end
    end

    def with_logging
      log = ExternalRequestLog.create!(<request info>)
      begin
        yield
      ensure
        log.update!(response: <response>)
      end
    end
```

Reply:
Ok, I like that!